File: src/main/java/com/healthcode/domain/HealthCodeType.java
package com.healthcode.domain;

/**
 * 健康码种类
 *
 * @author qianlei
 */
public enum HealthCodeType {
    /**
     * 绿码
     */
    GREEN("绿码"),
    /**
     * 黄码
     */
    YELLOW("黄码"),
    /**
     * 红码
     */
    RED("红码");
    private final String name;

    HealthCodeType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static HealthCodeType of(String value) {
        for (HealthCodeType type : values()) {
            if (type.name.equals(value)) {
                return type;
            }
        }
        return null;
    }
}

File: front/src/entity/HealthCodeType.ts
export enum HealthCodeType {
    GREEN = "GREEN",
    YELLOW = "YELLOW",
    RED = "RED"
}

export const healthCodeName = (type?: HealthCodeType): string => {
    switch (type) {
        case null:
            return "未填报"
        case HealthCodeType.GREEN:
            return "绿码"
        case HealthCodeType.RED:
            return "红码"
        case HealthCodeType.YELLOW:
            return "黄码"
        default:
            return "未填报"
    }
}